jit/baseline.ts

```diff
@@ -0,0 +1,1200 @@
+        case Bytecodes.DRETURN        : this.emitReturn(Kind.Double); break;
+        case Bytecodes.ARETURN        : this.emitReturn(Kind.Reference); break;
+        case Bytecodes.RETURN         : this.emitReturn(Kind.Void); break;
+        case Bytecodes.GETSTATIC      : cpi = stream.readCPI(); this.emitGetField(this.lookupField(cpi, opcode, true), true); break;
+        case Bytecodes.PUTSTATIC      : cpi = stream.readCPI(); this.emitPutField(this.lookupField(cpi, opcode, true), true); break;
+        case Bytecodes.GETFIELD       : cpi = stream.readCPI(); this.emitGetField(this.lookupField(cpi, opcode, false), false); break;
+        case Bytecodes.PUTFIELD       : cpi = stream.readCPI(); this.emitPutField(this.lookupField(cpi, opcode, false), false); break;
+        case Bytecodes.INVOKEVIRTUAL  : cpi = stream.readCPI(); this.emitInvoke(this.lookupMethod(cpi, opcode, false), opcode, stream.nextBCI); break;
+        case Bytecodes.INVOKESPECIAL  : cpi = stream.readCPI(); this.emitInvoke(this.lookupMethod(cpi, opcode, false), opcode, stream.nextBCI); break;
+        case Bytecodes.INVOKESTATIC   : cpi = stream.readCPI(); this.emitInvoke(this.lookupMethod(cpi, opcode, true), opcode, stream.nextBCI); break;
+        case Bytecodes.INVOKEINTERFACE: cpi = stream.readCPI(); this.emitInvoke(this.lookupMethod(cpi, opcode, false), opcode, stream.nextBCI); break;
+        case Bytecodes.NEW            : this.emitNewInstance(stream.readCPI()); break;
+        case Bytecodes.NEWARRAY       : this.emitNewTypeArray(stream.readLocalIndex()); break;
+        case Bytecodes.ANEWARRAY      : this.emitNewObjectArray(stream.readCPI()); break;
+        case Bytecodes.ARRAYLENGTH    : this.emitArrayLength(); break;
+        case Bytecodes.ATHROW         : this.emitThrow(stream.currentBCI); break;
+        case Bytecodes.CHECKCAST      : this.emitCheckCast(stream.readCPI()); break;
+        case Bytecodes.INSTANCEOF     : this.emitInstanceOf(stream.readCPI()); break;
+        ///*
+        case Bytecodes.MONITORENTER   : this.emitMonitorEnter(stream.nextBCI, this.pop(Kind.Reference)); break;
+        case Bytecodes.MONITOREXIT    : this.emitMonitorExit(this.pop(Kind.Reference)); break;
+        //case Bytecodes.MULTIANEWARRAY : genNewMultiArray(stream.readCPI()); break;
+        //*/
+        case Bytecodes.IFNULL         : this.emitIfNull(block, stream, Condition.EQ); break;
+        case Bytecodes.IFNONNULL      : this.emitIfNull(block, stream, Condition.NE); break;
+        ///*
+        //case Bytecodes.GOTO_W         : genGoto(stream.readFarBranchDest()); break;
+        //case Bytecodes.JSR_W          : genJsr(stream.readFarBranchDest()); break;
+        //case Bytecodes.BREAKPOINT:
+        //throw new CiBailout("concurrent setting of breakpoint");
+        //default:
+        //throw new CiBailout("Unsupported opcode " + opcode + " (" + nameOf(opcode) + ") [bci=" + bci + "]");
+        //}
+        //*/
```

Comment:
Here and above, some comments explaining why these cases are commented out would be useful.